File: archivist/src/main/java/com/zorroa/archivist/domain/DyHierarchySpec.java
package com.zorroa.archivist.domain;

import org.hibernate.validator.constraints.NotEmpty;

import javax.validation.constraints.NotNull;
import java.util.List;
import java.util.UUID;

/**
 * Created by chambers on 7/14/16.
 */
public class DyHierarchySpec {

    @NotNull
    private UUID folderId;

    @NotEmpty
    private List<DyHierarchyLevel> levels;

    public List<DyHierarchyLevel> getLevels() {
        return levels;
    }

    public DyHierarchySpec setLevels(List<DyHierarchyLevel> levels) {
        this.levels = levels;
        return this;
    }

    public UUID getFolderId() {
        return folderId;
    }

    public DyHierarchySpec setFolderId(UUID folderId) {
        this.folderId = folderId;
        return this;
    }
}

File: services/jvm/archivist/src/main/kotlin/com/zorroa/archivist/domain/DyHierarchySpec.kt
package com.zorroa.archivist.domain

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty
import java.util.UUID

@ApiModel("Dynamic Hierarchy Spec", description = "Data required to create a Dynamic Hierarchy (DyHi).")
class DyHierarchySpec(

    @ApiModelProperty("UUID of the Folder to apply the DyHi to.")
    val folderId: UUID,

    @ApiModelProperty("List of the DyHi Levels.")
    val levels: List<DyHierarchyLevel>,

    @ApiModelProperty("If true the DyHi will be generated.")
    val generate: Boolean = true

)
